cylc/flow/task_pool.py

```diff
@@ -1414,7 +1417,10 @@ def spawn_on_output(self, itask, output, forced=False):
                     tasks = [c_task]
 
                 for t in tasks:
-                    t.satisfy_me([itask.tokens.duplicate(task_sel=output)])
+                    t.satisfy_me(
+                        [itask.tokens.duplicate(task_sel=output)],
+                        getattr(itask.tdef, 'run_mode', RunMode.LIVE)
```

Comment:
This is returning the `run_mode` from the shared `TaskDef` object which is not necessarily the mode with which the task was actually submitted.